main.tf

```diff
@@ -222,28 +222,40 @@ locals {
   bucket_storage_class = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].storage_class : ibm_cos_bucket.cos_bucket1[*].storage_class
   s3_endpoint_public   = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].s3_endpoint_public : ibm_cos_bucket.cos_bucket1[*].s3_endpoint_public
   s3_endpoint_private  = var.encryption_enabled == true ? ibm_cos_bucket.cos_bucket[*].s3_endpoint_private : ibm_cos_bucket.cos_bucket1[*].s3_endpoint_private
```

Comment:
I think we usually require three properties for a bucket, the instance id (or guid, not sure which), the name and either the public or private endpoint.
So I think we should look at where buckets are consumed, maybe the replication example; and add either id or guid to the map so that consumers of the module get back complete bucket references that can be passed around modules and consumed easily.